src/rest/DataUnionEndpoints.js

```diff
@@ -248,270 +809,234 @@ export async function joinDataUnion(dataUnionContractAddress, secret) {
-        }
-        address = computeAddress(authKey)
-    }
-
-    const stats = await get(this, dataUnionContractAddress, `/members/${address}`)
-    if (stats.error || stats.earnings === '0') {
-        return {
-            total: BigNumber.ZERO, withdrawable: BigNumber.ZERO
-        }
-    }
-    const earningsBN = BigNumber.from(stats.earnings)
-
-    if (stats.withdrawableEarnings === '0') {
-        return {
-            total: earningsBN, withdrawable: BigNumber.ZERO
-        }
-    }
-    const withdrawableEarningsBN = BigNumber.from(stats.withdrawableEarnings)
-
-    const dataUnionContract = new Contract(dataUnionContractAddress, DataUnion.abi, provider || getDefaultProvider())
-    const withdrawnBN = await dataUnionContract.withdrawn(address)
-    const total = earningsBN.sub(withdrawnBN)
-    const withdrawable = withdrawableEarningsBN.sub(withdrawnBN)
-    return {
-        total, withdrawable
-    }
+export async function getMemberBalance(memberAddress, options) {
+    const address = parseAddress(this, memberAddress, options)
+    const duSidechain = await getSidechainContractReadOnly(this, options)
+    return duSidechain.getWithdrawableEarnings(address)
 }
 
-// TODO: filter? That JSON blob could be big
-export async function getMembers(dataUnionContractAddress) {
-    return getOrThrow(this, dataUnionContractAddress, '/members')
+export async function getTokenBalance(address, options) {
+    const a = parseAddress(this, address, options)
+    const tokenAddressMainnet = this.options.tokenAddress || options.tokenAddress
+    if (!tokenAddressMainnet) { throw new Error('tokenAddress option not found') }
+    const provider = this.getMainnetProvider()
+    const token = new Contract(tokenAddressMainnet, [{
+        name: 'balanceOf',
+        inputs: [{ type: 'address' }],
+        outputs: [{ type: 'uint256' }],
+        constant: true,
+        payable: false,
+        stateMutability: 'view',
+        type: 'function'
+    }], provider)
+    return token.balanceOf(a)
 }
 
-export async function getDataUnionStats(dataUnionContractAddress) {
-    return getOrThrow(this, dataUnionContractAddress, '/stats')
+/**
+ * Figure out if given mainnet address is old DataUnion (v 1.0) or current 2.0
+ * NOTE: Current version of streamr-client-javascript can only handle current version!
+ * @param {EthereumAddress} contractAddress
+ * @returns {number} 1 for old, 2 for current, zero for "not a data union"
+ */
+export async function getDataUnionVersion(contractAddress) {
+    const provider = this.getMainnetProvider()
+    const du = new Contract(contractAddress, [{
+        name: 'version',
+        inputs: [],
+        outputs: [{ type: 'uint256' }],
+        stateMutability: 'view',
+        type: 'function'
+    }], provider)
+    try {
+        const version = await du.version()
+        return +version
+    } catch (e) {
+        return 0
```

Comment:
Should double-check that the error isn't something unexpected e.g. network failure

Also ideally should use `err` or `error` for error variables, `e` is ungreppable.